crates/bifrost/src/providers/replicated_loglet/sequencer/mod.rs

```diff
@@ -297,7 +168,36 @@ impl<T: TransportConnect> Sequencer<T> {
             return Ok(LogletCommit::sealed());
         }
 
-        let next_write_offset = records.last_offset(self.next_write_offset)?.next();
+        if payloads.len() > self.max_in_flight_records_in_config {
+            let delta = payloads.len() - self.max_in_flight_records_in_config;
+            debug!(
+                "Resizing sequencer in-flight records capacity to allow admission for this batch. \
+                Capacity in configuration is {} and we are adding capacity of {} to it",
+                self.max_in_flight_records_in_config, delta
+            );
+            self.record_permits.add_permits(delta);
+        }
+
+        let len = u32::try_from(payloads.len()).expect("batch sizes fit in u32");
+        let permit = self
+            .record_permits
+            .clone()
+            .acquire_many_owned(len)
```

Comment:
What will happen here if the len is bigger than the entire Semaphore permit size

Reply:
I'm not sure what you mean by "entire semaphore permit size", If you are talking about the remaining capacity, then it'll block. If you're asking about `MAX_PERMITS`, then it's 61 bits, so u32 would definitely fit :)

Reply:
No, I don't mean remaining capacity. I mean if you created the Semaphore with 1000 permits, but the length of your batch is 2000.

Reply:
That's what the check in line 171 tries to address. It looks at the original provisioned capacity and tries to adjust but it's not a perfect solution as it doesn't account for previous resizes so it overshoots the capacity a little if this happens several times.